DigitalExcellence/dex-backend
# Feature/wizard

## Description

Implementation for the wizard, with a plugin-like system. 

### Finished:
- Implement adaptor pattern
- Create dynamic content for each data source adaptee. This allows to change for example the icon from a data source
- Reflection to search for data source adaptees
- Public and Authorized route for adaptees 
- Wizard pages so the front-end can figure out in what order the pages and what pages must get made visible. 
- Implement the oauth flow for the provider in a scalable 
- Create API calls from the adaptees to the external API (improving...)

## Type of change


## Checklist
This will get filled in later, most items on the list are partially and are worked at, so it can still change.


## Steps to Test or Reproduce

This will also get filled in later...

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Do something here...
2. Then run this test...
3. After that, you should see the result here

## Link to issue

Closes: #326
